### PR TITLE
Split package_repos resource and extend docker testing

### PR DESCRIPTION
**Pull request by enrico-usai**
### Description of changes
* Thanks to `cookbooks/aws-parallelcluster-test/recipes/docker_mock.rb` the actions are mocked so we can remove the virtualized check.
* gc_thresh_values and openssh are now tested on docker too.
* Permit to specify the action name when adding a resource as dep
* Split repos setup and update actions in the package_repos resource. Call the update actions for openssh test, requiring only apt_update call for Ubuntu 20.

### CODE REVIEW

File: kitchen.resources.yml [hunk 1]
```diff
@@ -36,7 +36,6 @@ suites:
 
 <% [
```
Comment (awslitvit) [inline kitchen.resources.yml]: Maybe we can remove this erb now?

File: test/recipes/controls/aws_parallelcluster_install/openssh_spec.rb [hunk 2]
```diff
@@ -25,10 +23,15 @@
   end
 
   files = %w(/etc/ssh/sshd_config /etc/ssh/ca_keys /etc/ssh/revoked_keys)
+  file_permission = if os.debian?
+                      '0644'
+                    else
+                      '0600'
+                    end
```
Comment (awslitvit) [inline test/recipes/controls/aws_parallelcluster_install/openssh_spec.rb:30]: Just out of curiosity: why?

File: recipes/add_dependencies.rb [hunk 3]
```diff
@@ -16,14 +17,22 @@
 #    dependencies:
 #      - recipe:aws-parallelcluster::test_dummy
 #      - recipe:aws-parallelcluster-install::directories
-#      - resource:ec2_udev_rules
+#      - resource:package_repos:update
```
Comment (awslitvit) [inline recipes/add_dependencies.rb]: Nitpick. I would keep both examples, the one with the default action and the one with the explicit action.

**Review by awslitvit — DISMISSED** [review]

File: kitchen.resources.yml (hunk 1 above)
Comment (enrico-usai) [reply kitchen.resources.yml]: Removed

File: recipes/add_dependencies.rb (hunk 3 above)
Comment (enrico-usai) [reply recipes/add_dependencies.rb]: Restored previous example too.

File: test/recipes/controls/aws_parallelcluster_install/openssh_spec.rb (hunk 2 above)
Comment (enrico-usai) [reply test/recipes/controls/aws_parallelcluster_install/openssh_spec.rb:30]: I verified it in the OS and this is the default. 
Not sure why in Ubuntu this is not aligned with RHEL but the root group has the possibility to read these files.

**Review by awslitvit — APPROVED** [review]

**Pull request merged**